src/DatabaseFactory.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Kros.KORM.Extensions.Asp.Properties;
+using Kros.Utils;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Kros.KORM.Extensions.Asp
+{
+    internal class DatabaseFactory : IDatabaseFactory
+    {
+        private static readonly ConcurrentDictionary<IServiceCollection, Dictionary<string, KormBuilder>> _builders =
+            new ConcurrentDictionary<IServiceCollection, Dictionary<string, KormBuilder>>();
+
+        private static Dictionary<string, KormBuilder> AddBuildersDictionary(IServiceCollection services)
+            => _builders.GetOrAdd(services, _ => new Dictionary<string, KormBuilder>());
+
+        internal static bool AddBuilder(IServiceCollection services, string name, KormBuilder builder)
+        {
+            Dictionary<string, KormBuilder> builders = AddBuildersDictionary(services);
+            if (builders.ContainsKey(name))
+            {
+                throw new ArgumentException(string.Format(Resources.DuplicateDatabaseName, name), nameof(name));
+            }
+            builders.Add(name, builder);
+            return builders.Count == 1;
```

Comment:
This is very confusing for me. When the method `Add..` returns a boolean, I assume that return `true` is if item added successfully. I looked at it for a long time, as long as I understood.